src/cljs/main/broadfcui/page/workspaces_list.cljs

```diff
@@ -83,31 +83,46 @@
   (->> workspaces (map func) (map count) (apply max)))
 
 (def ^:private access-types ["Project Owner" "Owner" "Writer" "Reader" "No Access"])
+(def ^:private access-types-defaults [true true true true true])
 (def ^:private access-predicates
   {"Project Owner" #(= "PROJECT_OWNER" (% "accessLevel"))
    "Owner" #(= "OWNER" (% "accessLevel"))
    "Writer" #(= "WRITER" (% "accessLevel"))
    "Reader" #(= "READER" (% "accessLevel"))
-   "No Access" #(= "NO ACCESS" (% "accessLevel"))
-   })
+   "No Access" #(= "NO ACCESS" (% "accessLevel"))})
 
-(def ^:private realm-types ["non-TCGA" "TCGA Open Access" "TCGA Protected Access"])
+(def ^:private dataset-types ["Un-published" "Published"])
+(def ^:private dataset-types-defaults [true false])
+(def ^:private dataset-predicates
+  (let [published? #(get-in % ["workspace" "attributes" "library:published"])]
+    {"Published" #(published? %)
+     "Un-published" #(not (published? %))}))
```

Comment:
`#(published? %)` can just be `published?`, and `#(not (published? %))` can be either `(comp not published?)` or `(complement published?)`